### ISSUE TITLE
Insufficient Fee

### ISSUE THREAD

**Issue by DietrichHoenen** [goatpig/BitcoinArmory#440]
Please Help.
I am trying to transfer all BTC from my Armory wallet, but I keep getting a Insufficient Fee error message.
If the system know that it is insufficient, then why not just let me know what would be sufficient?

How can I calculate what would be sufficient to make the transfer?

The error message suggested to attache the log. please find it below. 

[armorylog_20180113_1751.txt](https://github.com/goatpig/BitcoinArmory/files/1628522/armorylog_20180113_1751.txt)

**Comment by droark** [comment]
Which version are you running? The logs you posted are weird and don't have the usual information. If you're not on [0.96.4 RC1](https://github.com/goatpig/BitcoinArmory/releases/tag/v0.96.3.99) (Windows) or [RC2](https://github.com/goatpig/BitcoinArmory/releases/tag/v0.96.3.991) (Linux or macOS), please upgrade.

**Comment by DietrichHoenen** [comment]
Hi,
I was running 0.96.3, but now I have upgraded to 0.96.3.99 from your link.
Bitcoin Core Version 0.15.1

I tried again, Once it looked like it was going through, but I got the following message:
"Your Transaction failed to broadcast through the P2P layer but successfully broadcasted through the RPC. This can be a symptom of bad node connectivity to the Bitcoin network, or that your node is 
overwhelmed by network traffic. If you consistently get this warning, report to the developers for assistance with node maintenance."

The transaction never reached the blockchain, so I cleared all unconfirmed.

I tried again and then I got the Insuficient Fee error again. 
Fee 0.008 BTC (14.05 sat/B) 

How can I calculate what the correct fee should be?

[armorylog_20180114_1738.txt](https://github.com/goatpig/BitcoinArmory/files/1629537/armorylog_20180114_1738.txt)

**Comment by droark** [comment]
Sites like [this one](https://bitcoinfees.github.io/) can provide clues. That said, the latest Armory (RC3 is now out, by the way) should talk to Core in such a manner that Core provides a reasonably good fee estimate depending on how quickly you want things to confirm. You should be able to choose a fee that has a reasonable chance of confirming within, say, four blocks (or one if you're in a hurry). 14 sat/B shouldn't trip up anything in Core that I'm aware of, even though the fee's low enough that it'll probably be a long time (if ever) before it confirms.

**Comment by alexdriessen** [comment]
Hi, I have the same problem as Dietrich, I'm constantly getting the notification: "insufficient fee". This is strange, since I'm using 460 sat/B. I'm running Armory 0.96.3.922-beta-8a7f18a499 and Bitcoin Core 0.15.1. 

Could someone tell me what I'm doing wrong or missing?

[armorylog_20180118_2141.txt](https://github.com/goatpig/BitcoinArmory/files/1644390/armorylog_20180118_2141.txt)

**Comment by droark** [comment]
Upgrade to 0.96.4 RC3, delete your Armory DB, make sure ArmoryDB isn't running in the task manager (it gets stuck occasionally), and give it another spin. I don't know why but supposedly deleting the DB can fix the insufficient fee msg.

**Comment by alexdriessen** [comment]
Hi droark,

Thank you for you're quick reply. I did the Armory upgrade to 0.96.4 RC3, deleted the "databases" folder of Armory and let Armory rebuild the folder "databases". Unfortunately I'm still getting the same insufficient fee error.

**Comment by droark** [comment]
Okay. Just to be sure, you also confirmed that ArmoryDB wasn't running before starting up ArmoryDB again, correct?

**Comment by droark** [comment]
Yeah, a reboot would kill everything.

Hmmm. I'm not sure what's going on here. I've heard a few people mentioning the insufficient fee issue. I'm looking into something. In the meantime, can you run Help -> Clear All Unconfirmed and see if that helps?

**Comment by alexdriessen** [comment]
Unfortunately clearing the unconfirmed doesn't work neither.

**Comment by droark** [comment]
Yeah, I'm not sure offhand what's happening. It's Core that's complaining for whatever reason.

Can you do me a favor? Uninstall 0.15.1 (if that's what you're running) and install 0.14.2. You'll need to [start Core with the -reindex-chainstate option](https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-0.15.0.md), unfortunately, but I think you'll be fine after that. Once Core is synced, run Armory and try again. I have a vague theory and want to see if it works.

Thanks for your patience. I'll be busy the next couple of days but will get back to you when I can.

**Comment by alexdriessen** [comment]
BitCoin Core has not been complaining at all. I've tried the things you said also on a different computer (you never know if it helps) but no difference. I will try the installation of 0.14.2 and will let you know.

No problem, I have to thank you for your help, I'm the one that is stuck ;)

**Comment by droark** [comment]
Sorry, what I meant was that the "insufficient fee" message was coming from Core, hence my "complaining" comment. In any event, please keep us updated.

Also, would you happen to remember how you selected your fee? Did you set it manually or did you let Armory set it for you?

**Comment by droark** [comment]
Oh yeah, and one more thing. Do you have "listen=1" and "addnode=127.0.0.1" in your bitcoin.conf file? I don't think it'll matter but these are good entries to have nevertheless.

**Comment by alexdriessen** [comment]
I set the fee manually, I tried a fixed value (so that it is about 460 sat/B) and tried 460 sat/B, same error.

I can't find the bitcoin.conf file, it's not in users\...\roaming\bitcion\ nor anywhere on my SDD. Is it necessary for operation? I understand that you need to make that file according to: https://bitcointalk.org/index.php?topic=693211.0

Bitcoin core 0.14.2 is ready with synchronizing the blocks on the SSD in about 5 hours.

**Comment by alexdriessen** [comment]
Bitcoin Core is finished. I deleted the "databases" file of armory and then I run Armory 0.96.4 RC3 for the first time. I tried to sent some BTC with 300 sat/B and did not get any warning. But looking via blockchain.info, my transaction was not visible. So I rebooted Armory, checked again but still not visible on blockchain.info. Then I cleared all unconfirmed, rebooted Armory and tried to sent some BTC again. Now the "insufficient fee" error appeared again :(
